src/app/portlets/dot-edit-content/services/dot-edit-content-html.service.ts

```diff
@@ -181,24 +141,51 @@ export class DotEditContentHtmlService {
 
     private bindContainersEvents(): void {
         const addButtons = this.getEditPageDocument().querySelectorAll('.dotedit-container__add');
+        Array.from(addButtons).forEach((button: any) => {
+            const parent = button.parentElement;
+            const menuItems = parent.querySelectorAll('.dotedit-container__menu-item a');
 
-        addButtons.forEach((button) => {
-            this.bindButtonsEvent(button, 'add');
+            button.addEventListener('click', $event => {
+                this.closeContainersToolBarMenu(parent);
+                parent.classList.toggle('active');
+            });
+
+            Array.from(menuItems).forEach(menuItem => {
+                this.bindButtonsEvent(menuItem, 'add');
+            });
+        });
+    }
+
+    private bindWindowEvents(): void {
+        const doc = this.getEditPageDocument();
+
+        doc.addEventListener('click', $event => {
+            if (!$event.currentTarget.activeElement.classList.contains('dotedit-container__add')) {
+                this.closeContainersToolBarMenu();
+            }
+        });
+    }
+
+    private closeContainersToolBarMenu(activeElement?): void {
```

Comment:
I think activeElement is not a good name for this parameter

Reply:
suggestions ? it refers to the element that the user hit click on the DOM.